api/python/setup.py

```diff
@@ -70,24 +91,12 @@ def run(self):
         'requests_futures==1.0.0',
     ],
     extras_require={
-        'pyarrow': [
-            'numpy>=1.14.0',                # required by pandas, but missing from its dependencies.
-            'pandas>=0.19.2',
-            'pyarrow>=0.14.1',              # as of 7/5/19: linux/circleci bugs on 0.14.0
-        ],
-        'tests': [
-            'codecov',
-            'numpy>=1.14.0',                # required by pandas, but missing from its dependencies.
-            'pandas>=0.19.2',
-            'pyarrow>=0.14.1',              # as of 7/5/19: linux/circleci bugs on 0.14.0
-            'pytest<5.1.0',  # TODO: Fix pytest.ensuretemp in conftest.py
-            'pytest-cov',
-            'pytest-env',
-            'responses',
-            'tox',
-            'detox',
-            'tox-pytest-summary',
-        ],
+        'pyarrow': pyarrow_requires,
+        'tests': test_requires,
+        'all': [
+            *pyarrow_requires,
+            *test_requires,
```

Comment:
Technically, `tests` includes all of `pyarrow` dependencies, so this isn't really needed... But if you think it's cleaner this way, then sure.